lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -524,15 +524,83 @@ sub add_comment {
     my $job = $self->app->schema->resultset("Jobs")->find($self->param('testid'));
     return $self->reply->not_found unless $job;
 
+    # only logged in users can add comments
+    if (!$self->current_user) {
+        $self->flash('info', 'The comment couldn\'t be added because you\'re not logged in anymore');
+        return $self->redirect_to('test');
+    }
+
     my $rs = $job->comments->create(
         {
             text    => $self->param('text'),
             user_id => $self->current_user->id,
         });
+
```

Comment:
Adding lines for separation is certainly up to personal taste. Better just leave it as is. If you feel the methods get too big, split them rather than trying to separate by blank lines.